vmtest/vm.py

```diff
@@ -96,6 +97,57 @@
 "$BUSYBOX" hostname "$HOSTNAME"
 "$BUSYBOX" ip link set lo up
 
+run_command() {{
+    set +e
+    "$BUSYBOX" sh -c {command}
+    rc=$?
+    set -e
+    return $rc
+}}
+
+if "$BUSYBOX" [ -n "${{crashkernel:-}}" ]; then
+    # This isn't great: it's specific to x86-64, both by virtue of the syscall
+    # number and because kexec_file_load isn't implemented on many
+    # architectures, especially on older kernels. Also, embedding the script as
+    # a string here is ugly, but it avoids the need to locate yet another file.
+    "$PYTHON" -c '
+import ctypes
+import os
+import re
+import sys
+
+SYS_kexec_file_load = 320  # On x86-64.
+KEXEC_FILE_ON_CRASH = 2
+KEXEC_FILE_NO_INITRAMFS = 4
+
+syscall = ctypes.CDLL(None, use_errno=True).syscall
+syscall.restype = ctypes.c_long
+
+with open("/proc/cmdline", "rb") as f:
+    cmdline = f.read().rstrip(b"\n")
+    cmdline = re.sub(rb"(^|\s)crashkernel=\S+", b"", cmdline)
+    cmdline += b" nosmp nokaslr"
+
+with open(sys.argv[1], "rb") as kernel:
+    if syscall(
+        ctypes.c_long(SYS_kexec_file_load),
+        ctypes.c_int(kernel.fileno()),
+        ctypes.c_int(-1),
+        ctypes.c_ulong(len(cmdline) + 1),
+        ctypes.c_char_p(cmdline + b"\0"),
+        ctypes.c_ulong(KEXEC_FILE_ON_CRASH | KEXEC_FILE_NO_INITRAMFS),
+    ):
+        errno = ctypes.get_errno()
+        raise OSError(errno, os.strerror(errno))
+' "/lib/modules/$RELEASE/vmlinuz"
+    if run_command; then
+        "$BUSYBOX" echo 1 > /proc/sys/kernel/panic_on_oops
```

Comment:
What is this for? From my testing, it's not necessary for `sysrq-trigger` to work.

Reply:
Occasionally when the tests are being run in GitHub actions, `echo c > /proc/sysrq-trigger` fails to cause a kernel panic, but instead just triggers an Oops ([see here for an example](https://github.com/osandov/drgn/runs/4732642630?check_suite_focus=true#step:7:5950)). So in order to ensure a panic always occurs, I simply enable `panic_on_oops`.

Reply:
It looks like since torvalds/linux@8341f2f222d729688014ce8306727fdb9798d37e (in v5.0), `echo c > /proc/sysrq-trigger` uses `panic()` explicitly. For older versions, it stores to a `NULL` pointer, which indeed causes an oops. However, since torvalds/linux@cab8bd3410d448279e3bd0fbf96d31db0bf770fa (in v2.6.31, which is basically the dawn of time), the handler explicitly sets `panic_on_oops`.

I guess this doesn't hurt, but I also don't see how it'll help. Maybe the compiler is doing some weird optimizations because it sees the UB of dereferencing `NULL`. If you keep this, please also include a comment.